### PR TITLE
Use Go marshal API in the HR sample

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

**Comment by arv** [comment]
@aboodman Last commit shows marshal in action for samples/go/hr/

File: go/marshal/decode.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+// To unmarshal a Noms struct into a Go struct, Unmarshal matches incoming object
+// fields to the fields used by Marshal (either the struct field name or its tag).
+// Unmarshal will only set exported fields of the struct.
+//
+// If a Noms value is not appropriate for a given target type, or if a Noms number overflows the target type, Unmarshal returns a UnmarshalTypeMismatchError or OverflowError respectively
+//
+func Unmarshal(v types.Value, out interface{}) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			if _, ok := r.(runtime.Error); ok {
```
Comment (aboodman) [inline go/marshal/decode.go]: Interesting - where did you find this pattern?

Comment (aboodman) [inline go/marshal/decode.go]: (by which I mean how do you know it must be one of these three types?)

File: go/marshal/decode.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+				panic(r)
+			}
+			if s, ok := r.(string); ok {
+				panic(s)
+			}
+			err = r.(error)
```
Comment (aboodman) [inline go/marshal/decode.go]: This seems questionable. If you call some code and it panics with something that implements error, you will return it here. That doesn't seem like what you want. I think you should test for the particular errors you intend to panic with.

**Comment by aboodman** [comment]
seems reasonable to me

File: go/marshal/decode.go (hunk 2 above)
Comment (arv) [inline go/marshal/decode.go]: I didn't intend for this PR to be the review of the marshalling. Only to show the marshalling in action.

I'm going to go through the error handling code and update #2374 instead.

**Comment by arv** [comment]
@aboodman Now ready for review. PTAL.

**Comment by mikegray** [comment]
lgtm

**Comment by aboodman** [comment]
Better PR description needed.

**Comment by aboodman** [comment]
lgtm

**Pull request merged**